### ISSUE TITLE
Reinstate a camera offset option

### ISSUE THREAD

**Issue by YosypC** [opentrack/opentrack#1860]
Many version ago OpenTrack had the ability to implement an offset in pitch to account for different camera positions. Many have their cameras up on the monitor, often pitched down. Without additional corrections, the output will be skewed and will result in movements in axis which do no exist, for example: when pitched down, head Z movement will result in negative Y translation.

I've opened a new issue to continue the #1853 discussion.

**Comment by DaWelter** [comment]
I'm in favor of it. 

I guess it's on our/my todo list. Last year I made a draft for rotation correction only, included in this PR https://github.com/opentrack/opentrack/pull/1643 and didn't follow up on it. Maybe it's about time to revisit it. 

Ofc pull requests are welcome ;-)

**Comment by Cat-Lady** [comment]
Yes, please. The unwanted pitch during yaw is quite annoying, and it seems to be 100% related to camera position (above/under) and camera fixed pitch.

**Comment by Nels0** [comment]
Hi, I'm interested in this functionality too, or more specifically an arbitrary transform so that what's observed from the camera is remapped be relative to the screen's center.  Hoping to contribute code myself, but I'm still getting up to speed and getting build working.

A diagram because I'm not familiar with jargon, and excuse axis directions being inaccurate.
i.e: head moving in arrow direction will result in a translation with only a "screen (orange axes)" Z-component, whereas from the camera it would result in a translation with a Z and Y component.
and: monkey turning its head to the left resulting in rotation around the "screen" y-axis, whereas from the camera it would rotate around Y and X(?)

![image](https://github.com/user-attachments/assets/60f9beaa-734f-4560-8819-dd0617602171)

**Comment by IslandJohn** [comment]
This is the same #639, no? One or the other can probably be closed.

**Comment by YosypC** [comment]
> This is the same #639, no? One or the other can probably be closed.

It looks like so.

**Comment by YosypC** [comment]
Any news on the matter? I had to disable Y translation to account for the head zoom in many circumnstances.

**Comment by sthalik** [comment]
You guys won. I can no longer resist. It's on the shortlist for 2025.1.

**Comment by YosypC** [comment]
I'd be best to have some further explanation. Are the numbers' signs relative to the video source or the face? Is up negative or positive? Is it necessary to compile the X/Y/Z translation axis or just the pitch? I've spent around 30 minutes in trial and error for an imperfect result.

**Comment by Pieloth** [comment]
The easiest way to adjust the relative translation is to it in live mode:

- Start OT and tracking, place the OT window at center of your screen, and look at Game data, which should be at center of your screen
- open the OT Options page, Relative translation tab. Move it right so you have OT main + Options windows visible
- Now look towards screen center, to the Game data values, and use the 6 count boxes of the Relative translation tab to adjust the Game data values, so that they become 0
- When Game data values are 0 when looking at the center of the screen, then it should be correctly trimmed.

This is how I did, and it looks good on my side

**Comment by YosypC** [comment]
I still get a little "wave" effect on the Y axis. It's hard to debug without knowing what is necessary and what does what.

**Comment by DaWelter** [comment]
What @Pieloth said, and make sure to have no center pose set while you do that. Disable center on startup.

You can try moving your head forward and backward and tweak the offset angles so that mostly the z value is changing.

The offset angles are important. The translation is just added to your pose, so it's mostly "cosmetic".
